jdbc/src/main/java/org/apache/zeppelin/jdbc/SqlCompleter.java

```diff
@@ -217,11 +217,11 @@ public static Set<String> getSqlKeywordsCompletions(DatabaseMetaData meta) throw
     Set<String> completions = new TreeSet<>();
 
     if (null != meta) {
```

Comment:
```suggestion
    if (meta != null &&  meta.getDriverName() != null) {
```

Reply:
I see, therefore the additional null check `meta.getDriverName() != null`.
So the method should take the default keywords
```
String keywords =
            new BufferedReader(new InputStreamReader(
                    SqlCompleter.class.getResourceAsStream("/ansi.sql.keywords"))).readLine();
```
and return this set.

Reply:
However, exception is thrown once calling `getDriverName` , then `keywords` mentioned above cannot return and completer cannot work.  

Methods in the driver classes are implemented by other developers. We cannot ensure their actions, so put them in try-catch is more code-safe.

Reply:
Are you sure that the call `getDriverName()` throws an NPE? Looking at the code, I think that the call `getDriverName().replace('...')` throws the NPE because `getDriverName()` returns null.

I have looked at the implementation of the Hive driver. I hope I have found the right place. https://github.com/apache/hive/blob/489f6a20a5eedc1836854fd40255eb78aa967d99/jdbc/src/java/org/apache/hive/jdbc/HiveDriver.java#L264-L303

I don't see any possibility that an NPE is thrown. But an SQL exception can be thrown. To do this, `loadManifestAttributes();` would have to throw an IOException -> then the `SQL-Exception` is thrown.

Reply:
I didn't realize that `getDriverName()` can throw an `SQLException`, so I'm closing this suggestion.